File: src/main/java/edu/agh/neurons/Neuron.java
package edu.agh.neurons;

import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: krzysiek
 * Date: 03.04.13
 * Time: 20:32
 * To change this template use File | Settings | File Templates.
 */
public interface Neuron {

    public List<Double> weights();

    public double bias();

    public ActivationFunction activationFunction();
}

File: src/main/scala/edu/agh/neurons/scala/Neuron.scala
package edu.agh.neurons.scala

import activationFunctions.ActivationFunction

case class Neuron(var weights: Seq[Double],
                  bias: Double,
                  activationFunction: ActivationFunction) {

  def activate(values: Seq[Double]) =
    activationFunction.activate(values.zip(weights).map(el => el._1 * el._2).sum + bias)
}
